R/GetPostDeleteDefinition.R

```diff
@@ -253,6 +253,84 @@ postDefinition <- function(baseUrl, name, category, definition) {
   return(output)
 }
 
+#' Update definition \lifecycle{maturing}
+#' @details
+#' Update a definition in WebAPI. Currently only cohorts are supported.
+#' Takes the definition as a parameter and converts it to json. This is the full definition
+#' (i.e. including name and id fields)
+#' @template BaseUrl
+#' @template Category
+#' @param definition        An R list object containing the expression for the specification. This will be
+#'                          converted to JSON expression by function and posted into the WebApi.
+#' @examples
+#' \dontrun{
+#' definition <- getDefinition(id = 13242, baseUrl = "http://server.org:80/WebAPI", category = "cohort")
+#' definition$name <- "My new name for this"
+#' updateDefinition(definition, baseUrl, category = "cohort")
+#' }
+#' @export
+updateDefinition <- function(definition, baseUrl, category) {
+  baseUrl <- gsub("/$", "", baseUrl)
+  .checkBaseUrl(baseUrl)
+  arguments <- .getStandardCategories()
+  argument <- arguments %>% dplyr::filter(.data$categoryStandard == category)
+
+  errorMessage <- checkmate::makeAssertCollection()
+  checkmate::assertCharacter(definition$name, add = errorMessage)
+  checkmate::assertCharacter(category, add = errorMessage)
+  checkmate::assertNames(x = category, subset.of = arguments$categoryStandard)
+  checkmate::assertInteger(definition$id)
+  checkmate::reportAssertions(errorMessage)
+
+  # Check entity exists - this will throw a meaningful error if it doesn't
+  tryCatch(
```

Comment:
small change request for formatting - i think our OHDSI HADES convention is 

`tryCatch({`

not

```
tryCatch(
{
```